solana/gatekeeper-lib/src/service/GatekeeperService.ts

```diff
@@ -100,14 +138,29 @@ export class GatekeeperService {
       )
     );
 
-    await send(
-      this.connection,
-      transaction,
-      this.retries,
-      confirmOptions,
-      this.payer,
-      this.gatekeeperAuthority
-    );
+    try {
+      await send(
+        this.connection,
+        transaction,
+        confirmOptions,
+        this.payer,
+        this.gatekeeperAuthority
+      );
+    } catch (err) {
+      // The tx may have succeeded but the blockchain call timed out. Check on-chain.
+      const onChainToken: GatewayToken | null = await getGatewayToken(
+        this.connection,
+        gatewayTokenKey
+      );
+      if (onChainToken && onChainToken.state === State.ACTIVE) {
+        console.log(
+          "Send failed during Issue, but Active token is found on-chain. Returning success."
+        );
```

Comment:
would it not be better to have the return here and then there's no need for the 'else' statement?